src/App.js

```diff
@@ -1,13 +1,42 @@
 import React, { Component } from 'react'
+import changeColor from 'gen-random-color'
 import './App.css'
 
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      color: '#000000',
+      usedColors: []
+    }
+  }
+
+  repeatAction(func, num, delay) {
+    if (!num) 
+      return
+
+    const { usedColors } = this.state
+    const newColor = func(usedColors)
+    
+    this.setState({usedColors: [...usedColors, newColor], color: newColor})
+    
+    setTimeout(() => {this.repeatAction(func, num - 1, delay)}, delay)
+  }
+
+  componentWillMount() {
+    this.repeatAction(changeColor, 10, 1000)
```

Comment:
Yeah of course! I was getting at you trying to use a hash map so there would be an O(1) lookup to check if a color had been used already.